kedro/io/core.py

```diff
@@ -148,6 +159,23 @@ class AbstractDataset(abc.ABC, Generic[_DI, _DO]):
     need to change the `_EPHEMERAL` attribute to 'True'.
     """
     _EPHEMERAL = False
+    # Declares a class-level attribute that will store the initialization
+    # arguments of an instance. Initially, it is set to None, but it will
+    # hold a dictionary of arguments after initialization.
+    _init_args: dict[str, Any] | None = None
+
+    def __post_init__(self, call_args: dict[str, Any]) -> None:
```

Comment:
```suggestion
    def __post_init__(self, *args, **kwargs) -> None:
```

Btw, do we even need this method at all? Can't all of this be done by the decorator itself instead of delegating it to a separate method?

Reply:
Removed it after updates in the `__init_subclass__`